app/models/project.rb

```diff
@@ -860,13 +860,18 @@ def add_work_package(attributes = {})
     WorkPackage.new do |i|
       i.project = self
 
-      type_attribute = attributes.delete(:type) || attributes.delete(:type_id)
-
-      i.type = if type_attribute
-                 project.types.find(type_attribute)
+      type    = attributes.delete(:type)
+      type_id = if type && type.respond_to?(:id)
```

Comment:
@basiszwo I'm not that familiar with the `WorkPackage#add_work_package` API - I think this is a Redmine remnant -, but it's conceivable that a user passes in an attributes `Hash` with a `type: nil`, e.g. `type: Type.find(by_name: 'unknown')`